malcolm/modules/ADCore/parts/hdfwriterpart.py

```diff
@@ -247,7 +260,11 @@ def __init__(self, name, mri):
         self.uniqueid_offset = 0
         # The HDF5 layout file we write to say where the datasets go
         self.layout_filename = None  # type: str
+        self.runs_on_windows = runs_on_windows
         # Hooks
+        self.write_all_nd_attributes = BooleanMeta(
+            "Toggles wheteher all NDAttributes are written to file, or only those specified in the dataset",
+            writeable=True, tags=[Widget.CHECKBOX.tag(), config_tag()]).create_attribute_model()
```

Comment:
Can we add initial_write_all_ndattributes=True to __init__ and use it as the initial value of the attribute here?